### PR TITLE
Galaxy 2.0 update

### PR DESCRIPTION
**Pull request by chouseknecht**
- Improved documentation - man page, tweaks to galaxy rst, added galaxy section to configuration doc.
- quashed overloaded command options
- removed the helpful list of galaxy commands
- fixed handling of ignore_certs
- fixed bug in search command - terms were being added to the query in reverse order.

### CODE REVIEW

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 1]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+
+*--platforms*::
+
+Provide a comma separated list of Platforms on which to filter.
+
+*--author*::
+
+Specify the username of a Galaxy contributor on which to filter.
+
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling: `--ignore-certs`

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 2]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+
+OPTIONS
+~~~~~~~
+
+*-p* 'ROLES_PATH', *--roles-path=*'ROLES_PATH'::
+
+The path to the directory containing your roles. The default is the *roles_path* 
+configured in your *ansible.cfg* file (/etc/ansible/roles if not configured)
+
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling: `--ignore-certs`

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 3]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
 configured in your *ansible.cfg* file (/etc/ansible/roles if not configured)
 
 
+SEARCH
+------
+
+The *search* sub-command returns a filtered list of roles found at
+galaxy.ansible.com.
```
Comment (bcoca) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: not sure if its worth changing this to 'the api server' for when people are able to have private ones.

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 4]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+
+Override the default server https://galaxy.ansible.com.
+
+
+LOGIN
+-----
+
+The *login* sub-command is used to authenticate with galaxy.ansible.com. 
+Authentication is required to use the import, delete and setup commands.
+It will authenticate the user,retrieve a token from Galaxy, and store it
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: space needed between `user,` and `retrieve`

File: docsite/rst/galaxy.rst [hunk 5]
```diff
@@ -169,7 +181,9 @@ The search command will return a list of the first 1000 results matching your se
 
 .. note::
 
-    The format of results pictured here is new in Ansible 2.0.
+    The search command in Ansible 2.0 requires using the Galaxy 2.0 Beta site. Use the ``--server`` option to access 
+    `https://galaxy-qa.ansible.com <https://galaxy-qa.ansible.com>`_. You can also add a *server* definition in the [galaxy]
```
Comment (bcoca) [inline docsite/rst/galaxy.rst]: if the search command cannot be used yet, i would prefer we remove it and add it back when it is available.

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 6]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+
+If you do not wish to use your GitHub password, or if you have two-factor
+authentication enabled with GitHub, use the *--github-token* option to pass a
+personal access token that you create. Log into GitHub, go to Settings and
+click on Personal Access Token to create a token.
+
+OPTIONS
+~~~~~~~
+
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling `--ignore-certs`

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 7]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+authenticate with galaxy.ansible.com using the *login* subcommand.
+
+USAGE
+~~~~~
+
+$ ansible-galaxy import [options] github_user github_repo
+
+OPTIONS
+~~~~~~~
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling `--ignore-certs`

File: docsite/rst/intro_configuration.rst [hunk 8]
```diff
@@ -897,3 +897,19 @@ The normal behaviour is for operations to copy the existing context or use the u
...
+
+Galaxy Settings
+---------------
+
+The following options can be set in the [galaxy] section of ansible.cfg:
+
+server
+======
+
+Override the default Galaxy server value of https://galaxy.ansible.com.
```
Comment (bcoca) [inline docsite/rst/intro_configuration.rst]: we should point out here, that this is only useful if you have a hosted version of the galaxy web app, this won't work against private hosted repos which galaxy can still use for fetching roles.

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 9]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+  be added.
+
+* Provide your *Travis* user token as the secret. The token is not stored by
+  galaxy.ansible.com. A hash is created using github_user, github_repo
+  and your token. The hash value is what actually gets stored.
+
+OPTIONS
+~~~~~~~
+
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling `--ignore-certs`

File: docs/man/man1/ansible-galaxy.1.asciidoc.in [hunk 10]
```diff
@@ -145,6 +144,203 @@ The path to the directory containing your roles. The default is the *roles_path*
...
+
+USAGE
+~~~~~
+
+$ ansible-galaxy delete [options] github_user github_repo
+
+OPTIONS
+~~~~~~~
+
+*-c*, *--ingore-certs*::
```
Comment (trumant) [inline docs/man/man1/ansible-galaxy.1.asciidoc.in]: Spelling `--ignore-certs`

File: docsite/rst/galaxy.rst [hunk 11]
```diff
@@ -339,7 +359,9 @@ When you create your .travis.yml file add the following to cause Travis to notif
 
 .. note::
 
-    The setup command is only available in Ansible 2.0.
+    The setup command in Ansible 2.0 requires using the Galaxy 2.0 Beta site. Use the ``--server`` option to access 
+    `https://galaxy-qa.ansible.com <https://galaxy-qa.ansible.com>`_. You can also add a *server* definition in the [galaxy] 
+    section of your ansible.cfg file.
 
 
 List Travis Integrtions
```
Comment (trumant) [inline docsite/rst/galaxy.rst]: Spelling `Integrations`

File: lib/ansible/cli/galaxy.py [hunk 12]
```diff
@@ -50,50 +50,14 @@
...
-        "install":   "download a role into your roles path",
-        "list":      "enumerate roles found in your roles path",
-        "login":     "authenticate with Galaxy API and store the token",
-        "remove":    "delete a role from your roles path",
-        "search":    "query the Galaxy API",
-        "setup":     "add a TravisCI integration to Galaxy",
-    } 
-
     SKIP_INFO_KEYS = ("name", "description", "readme_html", "related", "summary_fields", "average_aw_composite", "average_aw_score", "url" )
+    VALID_ACTIONS = ("delete","import","info","init","install","list","login","remove","search","setup")
```
Comment (trumant) [inline lib/ansible/cli/galaxy.py]: Spacing between the tuple items seems to be the recommendation in the CODING_GUIDELINES.md

File: docsite/rst/galaxy.rst (hunk 5 above)
Comment (chouseknecht) [inline docsite/rst/galaxy.rst]: It is available now using https://galaxy-qa.ansible.com. We are doing a beta release of the new Galaxy site to be announced officially between now and Monday morning. That's the point of the note.

**Pull request merged**